### PR TITLE
Added silence trimming to seed generation

### PR DESCRIPTION
**Pull request by mortont**
Also fixed the case where a negative cut index could occur, which was not intended.

### CODE REVIEW

**Comment by mortont** [comment]
I'll squash those commits as well, if you'd prefer.

File: generate.py [hunk 1]
```diff
@@ -85,9 +85,12 @@ def create_seed(filename,
                 quantization_channels,
                 window_size=WINDOW):
     audio, _ = librosa.load(filename, sr=sample_rate, mono=True)
+    audio = audio_reader.trim_silence(audio)
 
     quantized = mu_law_encode(audio, quantization_channels)
-    cut_index = tf.size(quantized) + tf.constant(window_size) - tf.constant(1)
+    cut_index = tf.cond(tf.size(quantized) - tf.constant(window_size) < 0,
+            lambda: tf.constant(window_size),
```
Comment (ibab) [inline generate.py]: Shouldn't this be `lambda: tf.size(quantized)`, as we want to use the entire sample.
I think `window_size` would be outside of the sample in this case.

Comment (mortont) [inline generate.py]: Good catch, I'll change that.

File: generate.py [hunk 2]
```diff
@@ -85,9 +85,12 @@ def create_seed(filename,
...
+            lambda: tf.size(quantized) - tf.constant(window_size))
```
Comment (mortont) [inline generate.py]: This should also be changed to `lambda: tf.constant(window_size)`, come to think of it. That way the seed will be the first `window_size` of audio, or the entire input audio if it's smaller than `window_size`.

Comment (ibab) [inline generate.py]: Agreed!

**Comment by ibab** [comment]
Looks good, thank you!

**Pull request merged**